### ISSUE TITLE
loss: nan, grad_norm: nan

### ISSUE THREAD

**Issue by wujingxian1** [liyih/RCTrans#6]
After a few iterations of training, this issue came up. How can I fix it? If needed, I can provide the full training log. Thank you so much!

2025-04-18 21:58:31,047 - mmdet - INFO - Iter [10100/316440]	lr: 3.990e-04, eta: 7 days, 4:23:57, time: 2.125, data_time: 0.014, memory: 15378, frame_0_loss_cls: 0.0000, frame_0_loss_bbox: 0.0000, frame_0_d0.loss_cls: 0.0000, frame_0_d0.loss_bbox: 0.0000, frame_0_d1.loss_cls: 0.0000, frame_0_d1.loss_bbox: 0.0000, frame_0_d2.loss_cls: 0.0000, frame_0_d2.loss_bbox: 0.0000, frame_0_d3.loss_cls: 0.0000, frame_0_d3.loss_bbox: 0.0000, frame_0_d4.loss_cls: 0.0000, frame_0_d4.loss_bbox: 0.0000, frame_0_dn_loss_cls: 0.0000, frame_0_dn_loss_bbox: 0.0000, frame_0_d0.dn_loss_cls: 0.0000, frame_0_d0.dn_loss_bbox: 0.0000, frame_0_d1.dn_loss_cls: 0.0000, frame_0_d1.dn_loss_bbox: 0.0000, frame_0_d2.dn_loss_cls: 0.0000, frame_0_d2.dn_loss_bbox: 0.0000, frame_0_d3.dn_loss_cls: 0.0000, frame_0_d3.dn_loss_bbox: 0.0000, frame_0_d4.dn_loss_cls: 0.0000, frame_0_d4.dn_loss_bbox: 0.0000, frame_0_enc_loss_cls: nan, frame_0_enc_loss_bbox: nan, frame_0_enc_loss_iou: nan, frame_0_centers2d_losses: nan, frame_0_centerness_losses: nan, loss: nan, grad_norm: nan

**Comment by wujingxian1** [comment]
And Is this eta reasonable?

**Comment by liyih** [comment]
The eta seems too long, I use 8 A100 GPUs, it takes less than 20 hours. I guess the nan issue may related to small batchsize or the learning rate ( if the batch size decreases, the learning rate should be proportionally reduced).

**Comment by liyih** [comment]
你这个所有的loss都是0很奇怪，查查是因为nan造成的还是生产的label啥的有问题，还有就是你的NAN是在2d辅助任务上发生的（这是streampetr的baseline上的对应代码里的img_roi_head），你可以把这个辅助任务去掉看看还报错吗（相对于streampetr用辅助任务的输出做了spatial alignment module，我把这个内容去掉了），如果是辅助任务的问题，你可以看看streampetr的repo看看里面有没人遇到类似的问题

**Comment by liyih** [comment]
我记得我在尝试复现streampetr在vit backbone下效果的时候这个辅助任务好像确实很容易报nan，可能主要还是因为学习率和训练策略的问题。我这个代码的baseline用的是streampetr的，你可以去他的repo下找找解决方式

**Comment by liyih** [comment]
feel free to reopen this issue if you still have problem

**Comment by wujingxian1** [comment]
非常感谢您的回复！我改了学习率之后正常了
